src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -245,11 +246,11 @@ xrtGraphReset(xrtGraphHandle graph_hdl)
   }
   catch (const xrt_core::error& ex) {
     xrt_core::send_exception_message(ex.what());
-    return errno = ex.get();
+    return ex.get();
```

Comment:
There are two reasons I made this changes
1. Some of the API will call ioctl into zocl (like GMIO, reset), where the errno will be set by system
2. I am trying to unify the return behavior between PL kernel and AIE. So I referred to the style of xrt_kernel.cpp where the errno is not set

Reply:
I think we need to revisit this with better error handling in 21.1.   I believe errno should be set when error occurs and that xrt_kernel is not doing this because xrt::error was introduced after that.   Exceptions, e.g. throw xrt_core::error(-EINVAL, "...") can occur at many different places and not all places would set the errno, consolidating errno setting to one central place is better I think.   

Would it be a problem for the APIs to set errno per the exception value?  Presumably the exception value is the same as errno in the case where the error propagated from ioctl calls?

Reply:
If the exception value is the same, we should be fine. Otherwise, I think setting errno by API might overwrite the errno returned by driver, e.g.
xclSyncBO90;
ioctl(fd, ...); //(returns -EFAULT)
errno = -EINVAL;

Reply:
Yes. agreed.  Maybe errno should be reserved and managed by driver such that user space exceptions do not set errno at all.  I am not sure what is common practice?